drv/Functions.drv

```diff
@@ -230,7 +250,11 @@ public class FUNCTIONS {
 		public int hashCode() { synchronized(sync) { return function.hashCode(); } }
 
 		@Override
-		public boolean equals(Object o) { if (o == this) return true; synchronized(sync) { return function.equals(o); } }
+		public boolean equals(Object o) {
+			if (o == this) return true;
+			if (o instanceof SynchronizedFunction) synchronized (sync) { return function.equals(((SynchronizedFunction) o).function); }
```

Comment:
VERY DANGEROUS. Accesses a synchronized map without synchronization! 

In this cases, a look at the JDK code is a good idea...